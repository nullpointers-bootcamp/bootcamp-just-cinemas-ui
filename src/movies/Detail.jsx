import React from "react";
import "./detail.css";
class Detail extends React.Component {
  renderSynopsis() {
    const { synopsis } = this.props.movie;
    return (
      <div>
        <h4 className="synopsis-label">SYNOPSIS</h4>
        <p className="synopsis-content">{synopsis}</p>
      </div>
    );
  }

  renderStills() {
    const { stills } = this.props.movie;
    return (
      <div>
        <h4 className="stills-label">IMAGES</h4>
        <div className="stills-wrapper">
          {stills.map((still, index) => (
            <img
              className="still-item"
              alt={still}
              key={index}
              src={`https://s3.ap-south-1.amazonaws.com/twspicinemas/stills/${still.toLowerCase()}.jpg`}
            />
          ))}
        </div>
      </div>
    );
  }
  render() {
    const { name, slug } = this.props.movie;
    const imageUrl = `https://s3.ap-south-1.amazonaws.com/twspicinemas/banner/${slug.toLowerCase()}.jpg`;
    return (
      <div>
        <h1 className="name">{name}</h1>
        <img className="banner" alt={name} src={imageUrl} />
        {this.renderSynopsis()}
        {this.renderStills()}
      </div>
    );
  }
}

export default Detail;
