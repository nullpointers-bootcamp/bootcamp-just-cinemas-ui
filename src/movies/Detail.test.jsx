import React from "react";
import renderer from "react-test-renderer";
import Detail from "./Detail";

it("show movie detail information", () => {
  const movie = {
    id: 1,
    name: "Kabali",
    experiences: "RDX, Dolby Atmos, SUB",
    slug: "Kabali",
    synopsis: "Thalaiva's Crime thriller.",
    image: "kabali",
    stills: ["kabali", "kabali", "kabali"]
  };
  const tree = renderer.create(<Detail movie={movie} />).toJSON();
  expect(tree).toMatchSnapshot();
});
