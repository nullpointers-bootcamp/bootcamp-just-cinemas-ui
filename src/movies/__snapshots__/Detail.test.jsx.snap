// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show movie detail information 1`] = `
<div>
  <h1
    className="name"
  >
    Kabali
  </h1>
  <img
    alt="Kabali"
    className="banner"
    src="https://s3.ap-south-1.amazonaws.com/twspicinemas/banner/kabali.jpg"
  />
  <div>
    <h4
      className="synopsis-label"
    >
      SYNOPSIS
    </h4>
    <p
      className="synopsis-content"
    >
      Thalaiva's Crime thriller.
    </p>
  </div>
  <div>
    <h4
      className="stills-label"
    >
      IMAGES
    </h4>
    <div
      className="stills-wrapper"
    >
      <img
        alt="kabali"
        className="still-item"
        src="https://s3.ap-south-1.amazonaws.com/twspicinemas/stills/kabali.jpg"
      />
      <img
        alt="kabali"
        className="still-item"
        src="https://s3.ap-south-1.amazonaws.com/twspicinemas/stills/kabali.jpg"
      />
      <img
        alt="kabali"
        className="still-item"
        src="https://s3.ap-south-1.amazonaws.com/twspicinemas/stills/kabali.jpg"
      />
    </div>
  </div>
</div>
`;
